src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -23,8 +24,6 @@ public class HandleReferences : IIndexingWork
         private readonly DocumentsStorage _documentsStorage;
         private readonly IndexStorage _indexStorage;
 
-        private readonly Reference _reference = new Reference();
```

Comment:
Does it need to be removed?

Reply:
![image](https://user-images.githubusercontent.com/16625387/46915837-341d4400-cfba-11e8-8e5f-f2d97249dbde.png)

new Reference used in lambda expression instead (line 134 and 142)

Reply:
I understood that, but why we can't have one instance? When enumerating without a `.ToList()` that you removed it should work as it should.